### PR TITLE
Closes #184 Automatically updating the date of new blog posts

### LINKED ISSUE #184: Documentation:  PR Checklist or CI Actions for a date stamp update/reminder
### Let us know where something needs a refresh or put your idea here!

A posts date usually is the date it was started by the author.  We tend to forget to update the date once merged in and published.  The blog then sorts the posts by date and our most recent post is buried.

Two possible solutions:
1)  A CI Action could be used to automatically change the date stamp of the file to the date of the Merge?  
2)  A new checkbox in the PR Checklist for reviewers and creators of the post to check.

@StefanThoma maybe cool CI action to implement?

### PR DESCRIPTION
**Pull request by manciniedoardo**
Thank you for your Pull Request! We have developed this task checklist to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your blog post.

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update), and make sure the corresponding issue is linked in the Development section on the right hand side
- [x] Run the script from `CICD.R` line by line to first check the spelling in your post and then to make sure your code is compatible with our code-style. Address any incongruences by following the instructions in the file! 
- [x] Choose (possibly several) `tag(s)` or categories from the current list: ` c("Metadata", "SDTM", "ADaM", "TLG", "Shiny", "Community", "Conferences", "Submissions", "Technical")` for your blog post. If you cannot find anything that fits your blog post, propose a new tag to the maintainers! Note: if you use a tag not from this list, the "Check Post Tags" CICD pipeline will error. We occasionally tidy up all `tags` for consistency.
- [x] Add a short description for your blog post in the `description` field at the top of the markdown document.
- [x] Blog post is *short*, *personalized*, *reproducible* and *readable*
- [x] Add a disclaimer at the top of your post, if appropriate (example: **Disclaimer** 
This blog contains opinions that are of the authors alone and do not necessarily reflect the strategy of their respective organizations.)
- [x] Address all merge conflicts and resolve appropriately
- [x] Assign two of us (@bms63, @manciniedoardo, @StefanThoma, @kaz462) as reviewers in the PR.
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Comment by manciniedoardo** [comment]
The idea here is that posts now get created in a folder prefixed with `zzz_DO_NOT_EDIT`, and any time there is a push to main (e.g. when PRs are approved for new blog posts) any folders with that prefix get the folder name updated with the current date and the  post date within them is also updated.

File: R/create_blogpost.R [hunk 1]
```diff
@@ -12,7 +12,7 @@ print(allowed_tags)
 # Fill in the info, e.g.:
 create_post(
   post_name = "My Blog", # needs to be character vector (length 1)
-  post_date = "2024-12-01", # needs to be character vector (length 1)
+  post_date = "2024-12-01", # needs to be length 1character vector and will be auto-updated when your post is merged
```
Comment (manciniedoardo) [inline R/create_blogpost.R:15]:
```suggestion
  post_date = "2024-12-01", # needs to be length 1 character vector and will be auto-updated when your post is merged
```

File: R/readme.md [hunk 2]
```diff
@@ -10,3 +10,4 @@ Some of these files help in creating/developing blog-posts, others are used by o
 - `help_create_blogpost.R`: script containing the function(s) used by `create_blogpost.R`
 - `switch.R`: Used by CICD spellcheck workflow. 
 - `check_post_tags.R`: Used by Check-Post-Tags CICD workflow
+- `update_post_dates.R`: Used by Update-Post_Dates CICD workflow
```
Comment (manciniedoardo) [inline R/readme.md:13]:
```suggestion
- `update_post_dates.R`: Used by Update-Post-Dates CICD workflow
```

**Comment by StefanThoma** [comment]
> > Not a bad idea, but I don't understand how this will affect the main branch. I think it will update the name of the folder during the CICD run, but it will not be merged back into main, so this would still have to be done manually? Or am I missing something?
> 
> I tested this in a `cicd_testing` branch by putting `cicd_testing` instead of `main` in the workflow file. Basically it will be a push to `main` once the PR is merged, similar to how the vbump workflow functions.
> 
> Only question is whether we need to update the branch restrictions on `main` to enable this to work.

Hmm I think the part that says `On push main` or something just means that this will trigger the workflow to execute, but I don't think this will merge back into main. 
Or where do you think this will merge into main?

**Comment by manciniedoardo** [comment]
@StefanThoma I'm testing again with dummy branches - possible that I got this wrong during my first test

**Comment by manciniedoardo** [comment]
> @StefanThoma I'm testing again with dummy branches - possible that I got this wrong during my first test

You're right. I think we'd actually need to checkout main, do the update there, and then push?

**Comment by StefanThoma** [comment]
> > @StefanThoma I'm testing again with dummy branches - possible that I got this wrong during my first test
> 
> You're right. I think we'd actually need to checkout main, do the update there, and then push?

We also have to be mindful of the order, so that the date is adjusted before the publishing job runs, right?

**Comment by manciniedoardo** [comment]
> > > @StefanThoma I'm testing again with dummy branches - possible that I got this wrong during my first test
> > 
> > 
> > You're right. I think we'd actually need to checkout main, do the update there, and then push?
> 
> We also have to be mindful of the order, so that the date is adjusted before the publishing job runs, right?

Great point - maybe this workflow should be added as a dependency of the publishing job?

**Comment by manciniedoardo** [comment]
Ok, it should be fixed now - took a couple of attempts... Thanks @StefanThoma!
image
P.S. still need to be weary of order of execution with publishing job

**Comment by manciniedoardo** [comment]
@StefanThoma I updated the publishing job as well with a `needs: Update-post-dates` statement.

**Review by bms63 — COMMENTED** [review]
Whoa...

**Review by StefanThoma — APPROVED** [review]
LGTM!

**Review by gigikenneth — APPROVED** [review]

**Pull request merged**